### PR TITLE
feat(favorite page): #MED-249 add favorite page

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

Add favorite page and redirect favorite page on this new page and not on angular page

image

## Checklist tests

Check the filter, check the redirection

## Issue ticket number and link*

[MED-249](https://jira.support-ent.fr/browse/MED-249)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -56,14 +56,12 @@ export const Sidebar: React.FC<SidebarProps> = ({
...
-            name={`${t("mediacentre.sidebar.favorite")}`}
-            selected={location.pathname === "/favorites"}
-          />
-        </a>
+        <SidebarIcon
+          action={() => navigate("/favorites")}
+          icon={<StarIcon />}
+          name={`${t("mediacentre.sidebar.favorite")}`}
+          selected={location.pathname === "/favorites"}
+        />
```
Comment (niko-amosse) [inline frontend/src/components/sidebar/Sidebar.tsx:64]: on vire les <a> et on met le navigate à la place

File: frontend/src/routes/favorites/index.tsx
Comment (niko-amosse) [inline frontend/src/routes/favorites/index.tsx:1]: nouvelle page des favoris

File: frontend/src/routes/index.tsx
Comment (niko-amosse) [inline frontend/src/routes/index.tsx:1]: ajout de la route

File: frontend/src/routes/favorites/index.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+              </h1>
+            </div>
+          </div>
+          <div className="med-search-page-content-body">
+            <FilterLayout
+              resources={favoriteResourcesData}
+              setAllResourcesDisplayed={setAllResourcesDisplayed}
+            />
+            <InfiniteScrollList
+              redirectLink="/search"
```
Comment (Mattox0) [inline frontend/src/routes/favorites/index.tsx:79]: redirect LInk favorite (meme si pas utilisé)

File: frontend/src/routes/favorites/index.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+              allResourcesDisplayed={allResourcesDisplayed}
+              setAlertText={setAlertText}
+              refetchSearch={refetchFavorite}
```
Comment (Mattox0) [inline frontend/src/routes/favorites/index.tsx:82]: renome la fonction refetch aulieu de refetchSearch pour tout le monde pour que ce soit plus cohérent

File: frontend/src/routes/favorites/index.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  useEffect(() => {
+    if (favoriteResourcesData) {
+      setAllResourcesDisplayed(favoriteResourcesData);
+    }
+  }, [favoriteResourcesData]);
+
+  return (
+    <>
+      <MainLayout />
+      {alertText !== "" && (
```
[the remark this answers is not in the document]
Comment (niko-amosse) [reply frontend/src/routes/favorites/index.tsx:46]: incroyable !

File: frontend/src/routes/search/index.tsx [hunk 5]
```diff
@@ -1,231 +1,113 @@
...
+        "fr.openent.mediacentre.source.Moodle",
+        "fr.openent.mediacentre.source.Signet",
+      ],
+    };
+  };
+
+  const { allResources, refetchSearch } = useSearch(
+    createSearchBody(searchQuery),
+  );
+  const [allResourcesDisplayed, setAllResourcesDisplayed] = useState<
```
Comment (ThomasSevagen) [inline frontend/src/routes/search/index.tsx:43]: C'est quoi qui t'oblige à initialiser un state array à null? (il peut y avoir une bonne raison)
sinon tu initialise à tableau vide [] .
`const [allResourcesDisplayed, setAllResourcesDisplayed] = useState<Resource[]>([]);`
et tu peux utiliser `allRessourcesDisplayed.length` pour vérifier si le tableau est alimenté

**Review by Mattox0 — APPROVED** [review]

File: frontend/src/routes/search/index.tsx (hunk 5 above)
Comment (niko-amosse) [reply frontend/src/routes/search/index.tsx]: c'est pour notre loader ;)

**Pull request merged**